File: dmn-test-cases/signavio/dmn/dmn2java/expected/complex/dmn/example-credit-decision/CompareAgainstLendingThresholdRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "compareAgainstLendingThreshold"})
public class CompareAgainstLendingThresholdRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal compareAgainstLendingThreshold;

    public CompareAgainstLendingThresholdRuleOutput(boolean matched) {
        super(matched);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("compareAgainstLendingThreshold")
    public java.math.BigDecimal getCompareAgainstLendingThreshold() {
        return this.compareAgainstLendingThreshold;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("compareAgainstLendingThreshold")
    public void setCompareAgainstLendingThreshold(java.math.BigDecimal compareAgainstLendingThreshold) {
        this.compareAgainstLendingThreshold = compareAgainstLendingThreshold;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        CompareAgainstLendingThresholdRuleOutput other = (CompareAgainstLendingThresholdRuleOutput) o;
        if (this.getCompareAgainstLendingThreshold() != null ? !this.getCompareAgainstLendingThreshold().equals(other.getCompareAgainstLendingThreshold()) : other.getCompareAgainstLendingThreshold() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getCompareAgainstLendingThreshold() != null ? this.getCompareAgainstLendingThreshold().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", compareAgainstLendingThreshold='%s'", compareAgainstLendingThreshold));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-signavio/src/test/resources/dmn2java/exported/proto/expected/proto3/kotlin/dmn/example-credit-decision/CompareAgainstLendingThresholdRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "compareAgainstLendingThreshold"])
class CompareAgainstLendingThresholdRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("compareAgainstLendingThreshold")
    var compareAgainstLendingThreshold: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as CompareAgainstLendingThresholdRuleOutput
        if (if (this.compareAgainstLendingThreshold != null) this.compareAgainstLendingThreshold != other.compareAgainstLendingThreshold else other.compareAgainstLendingThreshold != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.compareAgainstLendingThreshold != null) this.compareAgainstLendingThreshold.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", compareAgainstLendingThreshold='%s'", compareAgainstLendingThreshold))
        result_.append(")")
        return result_.toString()
    }
}
